model.py

```diff
@@ -1,32 +1,34 @@
-
 from dataclasses import dataclass, field
 from typing import Optional, List
 from datetime import datetime
 from dateutil import parser
 
+
 @dataclass
 class Environment:
-	name: str
-	created_at: str
+    name: str
+    created_at: str
 
 
 @dataclass
 class Status:
-	name: str
-	status: bool
-	toggled_at: datetime = field(default=None)
+    name: str
+    status: bool
+    toggled_at: datetime = field(default=None)
+
+    def __post_init__(self):
+        if isinstance(self.toggled_at, str):
+            self.toggled_at = parser.isoparse(self.toggled_at)
```

Comment:
I don't like too much the __post_init__ method here and below. I think __init__ would be better. That's confusing in terms of types. Each initialization is going to be a string (because extracted from the json payload from Laika) and every time the post init method will cast it to datetime. Would you be ok if I change this as well @pvtmert ?

Reply:
AFAIK dataclass is native way to map json and it's added value is generating __init__ & it's friends (repr, str etc)

So it is mostly opinion, at the same time I had a script that's utilizing docker-py, which I can use this project there too.

At the same time it felt more "futureproof" and lacks necessity to maintain constructors by hand. If parent API adds new field, we'll just add it here. Order is kept by class itself.

Also JSON has no date or time specification, I can wrap it in try...except to make it more fool-proof (because there might be alternative APIs compatible with laika, even environment laika runs on can change it's date serialization format)

Reply:
You can also add __init__ with @dataclass . You still benefit of the other friends. I just don't find it right how it is because the generated init will always expect a `datetime` type because that's what you declared and you will mostly never passed a datetime object. In the end, you need to sanitize your init method with a __post_init__ that you need to maintain by hand :D